reconcile/dashdotdb_dora.py

```diff
@@ -0,0 +1,538 @@
+import functools
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from datetime import (
+    datetime,
+    timedelta,
+    timezone,
+)
+from typing import (
+    Any,
+    Optional,
+)
+
+import requests
+import yaml
+from psycopg2 import (
+    connect,
+    sql,
+)
+from sretoolbox.utils import threaded
+
+from reconcile import queries
+from reconcile.dashdotdb_base import (
+    LOG,
+    DashdotdbBase,
+)
+from reconcile.typed_queries.app_interface_vault_settings import (
+    get_app_interface_vault_settings,
+)
+from reconcile.typed_queries.saas_files import get_saas_files
+from reconcile.utils.github_api import GithubRepositoryApi
+from reconcile.utils.gitlab_api import GitLabApi
+from reconcile.utils.secret_reader import create_secret_reader
+
+QONTRACT_INTEGRATION = "dashdotdb-dora"
+
+
+@dataclass
+class DeploymentDBSecret:
+    path: str
+    field: str
+    q_format: Optional[str]
+    version: Optional[int]
+
+
+@dataclass(eq=True, frozen=True)
+class AppEnv:
+    app_name: str
+    env_name: str
+
+
+@dataclass
+class Deployment:
+    trigger_reason: str
+    deployment_time: datetime
+
+
+@dataclass(eq=True, frozen=True)
+class SaasTarget:
+    app_name: str
+    env_name: str
+    path: str
+    resource_template: str
+    namespace: str
+    pipeline: str
+
+    @property
+    def app_env(self) -> AppEnv:
+        return AppEnv(self.app_name, self.env_name)
+
+
+@dataclass(eq=True, frozen=True)
+class RepoChanges:
+    repo_url: Optional[str]
+    ref_from: Optional[str]
+    ref_to: Optional[str]
+
+
+@dataclass(eq=True, frozen=True)
+class SummaryKey:
+    rt_name: str
+    target_ns: str
+
+
+@dataclass(eq=True, frozen=True)
+class SummaryEntry:
+    repo_url: str
+    target_ref: str
+
+
+class DeploymentDB:
+    def __init__(
+        self,
+        host: str,
+        port: str,
+        name: str,
+        user: str,
+        password: str,
+    ):
+        self.conn = connect(
+            "host={} port={} dbname={} user={} password={}".format(
```

Comment:
Please use the f-string syntax. 🙇

Reply:
or use kwargs if no special requirement, according to doc here https://www.psycopg.org/docs/module.html#psycopg2.connect

> The connection parameters can be specified as a [libpq connection string](https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING) using the dsn parameter:
conn = psycopg2.connect("dbname=test user=postgres password=secret")
or using a set of keyword arguments:
conn = psycopg2.connect(dbname="test", user="postgres", password="secret")

Reply:
:this:

Reply:
done with kwargs, I don't know why I didn't see this before! thanks!